test/integrational/distribution.js

```diff
@@ -588,96 +588,53 @@ describe('Distribution', async function() {
-                                balancesDelta.usdc[order.redeemerWallet]
-                            ).to.be.bignumber.equal(toBN('0'))
-                            continue
-                        }
-                        expect(
-                            balancesDelta.blend[order.redeemerTenderAddress]
-                        ).to.be.bignumber.equal(
-                            order.amount.neg()
-                        )
-                        expect(
-                            balancesDelta.usdc[order.redeemerWallet]
-                        ).to.be.bignumber.equal(
-                            blendToUsdc(order.amount, order.price)
-                        )
-                    }
 
-                    expect(
-                        await scenario.remainingAllowance()
-                    ).to.be.bignumber.equal(
-                        undistributedUsdc
+                    await expectRevert(
+                        scenario.run(),
+                        'Burning from regular addresses is not allowed'
                     )
                 }
             )
         })
 
-        it('Partially executes underfunded orders', async function() {
+        it('Fails if there is an underfunded order', async function() {
             await testProperty(
                 () => prepareScenario(RunStrategy.ONE_BATCH),
-                async ({scenario, snapshot}) => {
-                    // Here we simply add some big amount to orders. Since
-                    // tender balance does not change, we successfully simulate
-                    // an underfunding situation. We remember the balance of the
-                    // tender address, though, to check whether the order has
-                    // been executed successfully.
-                    for (let order of scenario.orders) {
-                        order.partialAmount =
-                            snapshot.blend[order.redeemerTenderAddress]
-                        order.amount = order.amount.add(toBN('100000000'))
-                    }
-
-                    await scenario.run()
-                    const balancesDelta = await scenario.balancesDelta(snapshot)
-
-                    for (let order of scenario.orders) {
-                        expect(
-                            balancesDelta.blend[order.redeemerTenderAddress]
-                        ).to.be.bignumber.equal(
-                            order.partialAmount.neg()
-                        )
-                        expect(
-                            balancesDelta.usdc[order.redeemerWallet]
-                        ).to.be.bignumber.equal(
-                            blendToUsdc(order.partialAmount, order.price)
-                        )
-                    }
+                async ({scenario}) => {
+                    const underfundedIdx =
+                        getRandomInt(0, scenario.orders.length - 1)
+                    scenario.orders[underfundedIdx].amount =
```

Comment:
what if tender address contained much more than sum amount of orders?

Reply:
You mean if I have tender address with N blend tokens, and I want to participate in tender offer, where the sum of all order is, like, 20, what would happen?

1) In case N ~ MAX_UINT256, I expect an overflow due to `price * amount` computation
2) In case N < MAX_UINT256 / 10000, the balance should just decrease by the orders sum (+fee but here fee is equal to zero)

Reply:
I mean, it seems like underfunded case won't trigger

Reply:
Why should it? If the tender address balance is enough to execute the order, it should actually execute the order

Reply:
Or do you mean the **order amount** is much higher than the tender balance?

i.e. not `amount = amount + 1` but `amount = MAX_UINT256 / 10000`?